plugins/BEdita/Core/src/Model/Behavior/UniqueNameBehavior.php

```diff
@@ -54,9 +55,47 @@ public function uniqueName(EntityInterface $entity)
         $config = $this->config();
         $uname = $entity->get('uname');
         if (empty($uname) && empty($entity->get('id'))) {
-            $uname = $config['prefix'] . Text::slug($entity->get($config['sourceField']), $config['replacement']);
+            $source = $entity->source();
+            $regenerate = false;
+            do {
+                $uname = $this->generateUniqueName($entity, $config, $regenerate);
+            } while ($regenerate = $this->uniqueNameExists($source, $uname));
         }
-        $entity->set('uname', strtolower($uname));
+        $entity->set('uname', $uname);
+    }
+
+    /**
+     * Generate unique name string from $config parameters.
+     * If $regenerate parameter is true, random hash is added to uname string.
+     *
+     * @param \Cake\Datasource\EntityInterface $entity The entity to save
+     * @param array $config parameters to create unique name
+     * @param bool $regenerate if true it adds hash string to uname
+     * @return string uname
+     */
+    public function generateUniqueName(EntityInterface $entity, array $config, $regenerate = false)
+    {
+        $uname = $config['prefix'] . Text::slug($entity->get($config['sourceField']), $config['replacement']);
+        if ($regenerate) {
+            $hash = sha1(md5($uname));
```

Comment:
Ok using this hash, but it adds a 40 string hash without a dash.
I would fetch less characters, say first or last 6 or add a config to choose hash length with a low default